src/autoscaler/metricscollector/cmd/metricscollector/metricscollector_suite_test.go

```diff
@@ -42,31 +43,24 @@ var (
 	eLock          *sync.Mutex
 	httpClient     *http.Client
 	consulRunner   *consulrunner.ClusterRunner
+	setupParams    arguments
 )
 
 func TestMetricsCollector(t *testing.T) {
 	RegisterFailHandler(Fail)
 	RunSpecs(t, "MetricsCollector Suite")
 }
 
+type arguments struct {
+	McBuildPath  string
+	CcNoaaUaaURL string
+	Lock         *sync.Mutex
+}
+
 var _ = SynchronizedBeforeSuite(func() []byte {
 	mc, err := gexec.Build("autoscaler/metricscollector/cmd/metricscollector", "-race")
 	Expect(err).NotTo(HaveOccurred())
 
-	return []byte(mc)
-}, func(pathsByte []byte) {
-
-	consulRunner = consulrunner.NewClusterRunner(
-		consulrunner.ClusterRunnerConfig{
-			StartingPort: 9001 + GinkgoParallelNode()*consulrunner.PortOffsetLength,
-			NumNodes:     1,
-			Scheme:       "http",
-		},
-	)
-	consulRunner.Start()
-	consulRunner.WaitUntilReady()
-	mcPath = string(pathsByte)
-
 	ccNOAAUAA = ghttp.NewServer()
```

Comment:
why setting up the ccNOAAUAA in primary node?  We can do it in each node, and have different app IDs for different node so that  test cases do not share any ccNoaaUAA server and data in metricscollector database